### ISSUE TITLE
Custom Color for Bars & Columns

### ISSUE THREAD

**Issue by arpit0515** [PHPOffice/PhpSpreadsheet#1433]
This is:

### What is the expected behavior?
I am trying to custom color the bars of the bar chart & column chart which is being exported using phpoffice/phpspreadsheet. I have used multiple codes and ways which are mentioned in the pie chart example code but it is not working.

### What is the current behavior?
default color scheme is being applied to the bar charts

### What are the steps to reproduce?
create any bar chart using phpoffice/phpspreadsheet and try to change the color either in the dataserieslabels or dataseriesvalues. Nothing is happening. 

Please provide a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

```php
<?php

require __DIR__ . '/vendor/autoload.php';

// Create new Spreadsheet object
$spreadsheet = new \PhpOffice\PhpSpreadsheet\Spreadsheet();

// add code that show the issue here...

$dataSeriesLabels = [
    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_STRING, 'Data!$C$1', null, 1, [], null, ["#CCCCCC"]), // 'Budget'
    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_STRING, 'Data!$D$1', null, 1, [], null, ["#D6D6D6"]), // 'Forecast'
    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_STRING, 'Data!$E$1', null, 1, [], null, ["#B8292F"]), // 'Actual'
    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_STRING, 'Data!$F$1', null, 1, [], null, ["#EB8500"]), // 'Actual'
];

$dataSeriesValues = [
    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, 'Data!$C$2:$C$5', null, 4, [], null, array('#cccccc', '#00abb8', '#b8292f', '#eb8500')),
    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, 'Data!$D$2:$D$5', null, 4, [], null, array('#cccccc', '#00abb8', '#b8292f', '#eb8500')),
    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, 'Data!$E$2:$E$5', null, 4, [], null, array('#cccccc', '#00abb8', '#b8292f', '#eb8500')),
    new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, 'Data!$F$2:$F$5', null, 4, [], null, array('#cccccc', '#00abb8', '#b8292f', '#eb8500')),
];
```

I was actually able to find a fix for this, have updated this from a feature request to a bug. Please share an update when the regular code is up and working, this is just an alternative for the ones who are in urgent need of applying custom color codes to bar charts. This is however not working fine for the trendline charts. I am still exploring options to change the color of lines in trendline charts

```php
<?php 

$dataSeriesLabels = [
     new \PhpOffice\PhpSpreadsheet\Chart\DataSeriesValues('String', 'Data!$B$1', null, 1, [], null, "FF4040"),
     new \PhpOffice\PhpSpreadsheet\Chart\DataSeriesValues('String', 'Data!$C$1', null, 1, [], null, "29DCFF"),
     new \PhpOffice\PhpSpreadsheet\Chart\DataSeriesValues('String', 'Data!$D$1', null, 1, [], null, "FFCB2D"),
     new \PhpOffice\PhpSpreadsheet\Chart\DataSeriesValues('String', 'Data!$E$1', null, 1, [], null, "96979C"),
];

```

### Which versions of PhpSpreadsheet and PHP are affected?
 I am working on PHP v7

**Comment by erbarratt** [comment]
This is still an issue. I tried to use:

```
$chartDataSeries = [];
							
							//colors need formatting correctly to be accepted
								$colors = [];
								foreach($_SESSION['graphs'][$g]['colors'] as $colorHere){
									
									$colors[] = mb_strtolower(str_replace('#', '', $colorHere));
									
								}
							
							//pie data series different
								if($_SESSION['graphs'][$g]['type'] == 'donut' || $_SESSION['graphs'][$g]['type'] == 'pie'){
									
									$dataSeriesHere = new \PhpOffice\PhpSpreadsheet\Chart\DataSeriesValues(
										\PhpOffice\PhpSpreadsheet\Chart\DataSeriesValues::DATASERIES_TYPE_NUMBER, 
										'Data!$B$4:$B$'.$tableRow, 
										null, 
										($tableRow - 3), 
										[], 
										null, 
										$colors
									);
									
									$dataSeriesHere->setFillColor($colors);
									
									$chartDataSeries = [$dataSeriesHere];

							//all other chart types
								} else {
									
									//4 is the row that the series labels start
										$colorIndex = 0;
										for ($x = 4; $x <= $tableRow; $x++){
											
											$dataSeriesHere = new \PhpOffice\PhpSpreadsheet\Chart\DataSeriesValues(
												\PhpOffice\PhpSpreadsheet\Chart\DataSeriesValues::DATASERIES_TYPE_NUMBER, 
												'Data!$B$'.$x.':$'.$maxCol.'$'.$x, 
												null, 
												$catCount, 
												[], 
												null, 
												$colors[$colorIndex]
											);
											
											$dataSeriesHere->setFillColor($colors[$colorIndex]);
											
											$chartDataSeries[] = $dataSeriesHere;
											
											$colorIndex++;
											
										}
									
								}

```

**Comment by oleibman** [comment]
Addressed by PR #2898 in July 2022.
